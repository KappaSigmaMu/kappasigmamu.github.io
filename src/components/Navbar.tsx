import { Button, Container, Nav, Navbar as RBNavbar } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import { useAccount } from '../account/AccountContext'
import { fetchAccounts } from '../helpers/fetchAccounts'
import KappaSigmaMu from '../static/kappa-sigma-mu-logo.svg'
import { AccountSelector } from './AccountSelector'
import { SocialIcons } from './SocialIcons'

const Navbar = ({
  showAccount=false,
  showBrandIcon=false,
  showExploreButton=false,
  showSocialIcons=false,
}: NavRouteProps) => (
  <RBNavbar className="mt-4">
    <Container>
      <Nav>{showBrandIcon ? <NavbarBrand /> : <BrandPlaceholder />}</Nav>
      <Nav className="align-items-center align-self-center">
        {showExploreButton ? <Nav.Link to="/explore" as={Link}>Explore</Nav.Link> : <></>}
        &nbsp;
        {showSocialIcons ? <SocialIcons /> : <></>}
        &nbsp;
        {showAccount ? <AccountNavbar /> : <></>}
      </Nav>
    </Container>
  </RBNavbar>
)

const BrandPlaceholder = () => (
  <div style={{ height: 82, width: 106 }}></div>
)

const NavbarBrand = () => (
  <RBNavbar.Brand as={Link} to="/">
    <img width={90} src={KappaSigmaMu} alt="KappaSigmaMu Logo" />
  </RBNavbar.Brand>
)

const AccountNavbar = () => {
  const { activeAccount, setActiveAccount, setAccounts } = useAccount()

  return activeAccount
    ? <AccountSelector />
    : (
      <Button variant="outline-secondary" onClick={() => fetchAccounts(setAccounts, setActiveAccount)}>
        Connect Wallet
      </Button>
    )
}

export { Navbar }
