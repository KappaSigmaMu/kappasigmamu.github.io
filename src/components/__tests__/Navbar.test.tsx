import { BrowserRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import { Navbar } from '../Navbar'

describe('NavBar', () => {
  it('renders it with showAccount', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar showAccount />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showBrandIcon', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar showBrandIcon />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showExploreButton', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar showExploreButton />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showSocialIcons', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar showSocialIcons />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })
})
