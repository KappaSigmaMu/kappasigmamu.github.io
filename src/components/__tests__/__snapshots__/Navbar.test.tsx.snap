// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders it with showAccount 1`] = `
<nav
  className="pt-4 navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        style={
          Object {
            "height": 82,
            "width": 106,
          }
        }
      />
    </div>
    <div
      className="align-items-center align-self-center navbar-nav"
      onKeyDown={[Function]}
    >
       
       
      <div
        className="spinner-border text-primary"
      />
    </div>
  </div>
</nav>
`;

exports[`NavBar renders it with showBrandIcon 1`] = `
<nav
  className="pt-4 navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="KappaSigmaMu Logo"
          src="kappa-sigma-mu-logo.svg"
          width={90}
        />
      </a>
    </div>
    <div
      className="align-items-center align-self-center navbar-nav"
      onKeyDown={[Function]}
    >
       
       
    </div>
  </div>
</nav>
`;

exports[`NavBar renders it with showExploreButton 1`] = `
<nav
  className="pt-4 navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        style={
          Object {
            "height": 82,
            "width": 106,
          }
        }
      />
    </div>
    <div
      className="align-items-center align-self-center navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/explore"
        onClick={[Function]}
      >
        Explore
      </a>
       
       
    </div>
  </div>
</nav>
`;

exports[`NavBar renders it with showSocialIcons 1`] = `
<nav
  className="pt-4 navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        style={
          Object {
            "height": 82,
            "width": 106,
          }
        }
      />
    </div>
    <div
      className="align-items-center align-self-center navbar-nav"
      onKeyDown={[Function]}
    >
       
      <div
        className="me-2"
      >
        <a
          className="sc-jSMfEi"
          href="https://discord.gg/9AWjTf8wSk"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Discord Logo"
            src="discord-logo.svg"
          />
        </a>
         
        <a
          className="sc-jSMfEi"
          href="https://matrix.to/#/!BUmiAAnAYSRGarqwOt:matrix.parity.io?via=matrix.parity.io"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Element Logo"
            src="element-logo.svg"
          />
        </a>
         
        <a
          className="sc-jSMfEi"
          href="https://twitter.com/kusamanetwork"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Twitter Logo"
            src="twitter-logo.svg"
            style={
              Object {
                "color": "red",
              }
            }
          />
        </a>
      </div>
       
    </div>
  </div>
</nav>
`;
